tests/models/xumx_test.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+import torch
+
+from asteroid.models import XUMX
+
+sources = [
+    ["bass", "drums", "vocals", "other"],
+    ["vocals", "drums", "bass"],
+    ["vocals", "other"],
+    ["drums"],
+]
+
+
+@pytest.mark.parametrize("nb_channels", (1, 2))
+@pytest.mark.parametrize("sources", sources)
+@pytest.mark.parametrize("bidirectional", (True, False))
+@pytest.mark.parametrize("spec_power", (1, 2))
+@pytest.mark.parametrize("return_time_signals", (True, False))
+@pytest.mark.parametrize(
+    "data",
+    (
+        torch.rand(4, 2, 44100, requires_grad=False),
+        torch.rand(1, 2, 2 * 44100, requires_grad=False),
+        torch.rand(8, 2, 100000, requires_grad=False),
+    ),
+)
+def test_forward(nb_channels, sources, bidirectional, spec_power, return_time_signals, data):
+    x_umx = XUMX(
+        window_length=4096,
+        input_mean=None,
+        input_scale=None,
+        nb_channels=nb_channels,
+        hidden_size=256,
+        in_chan=4096,
+        n_hop=1024,
+        sources=sources,
+        max_bin=1000,
+        bidirectional=bidirectional,
+        sample_rate=44100,
+        spec_power=spec_power,
+        return_time_signals=return_time_signals,
+    )
+    x_umx = x_umx.eval()
+    with torch.no_grad():
+        x_umx(data)
+
+
+def test_get_model_args():
+    sources_tmp = {"src_0": "vocals"}
```

Comment:
Should this be a dict or a list?

Reply:
Sorry for my careless mistake! This part is old one, and should be type of list.

Reply:
Resolved by #490 fa6ae66